### PR TITLE
Rails 7.1 Update

### PR DESCRIPTION
**Pull request by tsmartt**
The new renovate checker will open 93 PRs, this is work to get ahead of that.

### CODE REVIEW

File: test/controllers/publishers/omniauth_callbacks_controller_test.rb [hunk 1]
```diff
@@ -785,6 +785,7 @@ def auth_hash(options = {})
       request_login_email(publisher: publisher)
       url = publisher_url(publisher, token: publisher.reload.authentication_token)
       get(url)
+      puts response.body
```
Comment (jlbyrne) [inline test/controllers/publishers/omniauth_callbacks_controller_test.rb:788]: nitpick- I don't think you meant to leave this in here.

**Review by jlbyrne — APPROVED** [review]
LGTM!

File: test/controllers/publishers/omniauth_callbacks_controller_test.rb (hunk 1 above)
Comment (tsmartt) [reply test/controllers/publishers/omniauth_callbacks_controller_test.rb:788]: thanks!

File: config/environments/test.rb [hunk 2]
```diff
@@ -40,12 +47,12 @@
   }
 
   # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
-  # config.action_controller.perform_caching = false
-  # config.cache_store = :null_store
+  config.consider_all_requests_local = true
```
[the remark this answers is not in the document]
Comment (tsmartt) [reply config/environments/test.rb:50]: this is for the test env and I don't know why semgrep doesn't get this, it's the same in basically every rails app

[the remark this answers is not in the document]
Comment (thypon) [reply config/environments/test.rb:50]: We can have this file ignored in a `.semgrepignore` file

File: Gemfile [hunk 3]
```diff
@@ -98,15 +98,15 @@ gem "premailer-rails", "~> 1.12.0", require: false
 gem "public_suffix", "~> 5.0.1"
 
 # Puma as app server
-gem "puma", "~> 6.0.2"
+gem "puma", "~> 6.3.0"
 
 # Make cracking a little bit harder
-gem "rack-attack", "~> 6.6.1"
+gem "rack-attack", github: "rocketreferrals/rack-attack", branch: "allow-rack-3"
```
Comment (thypon) [inline Gemfile:104]: can't we use that branch in `rack/rack-attack`?

File: Gemfile [hunk 4]
```diff
@@ -223,11 +224,11 @@ group :development, :test do
   gem "bundler-audit", require: false
   gem "capybara"
   gem "minitest"
-  gem "minitest-rails"
+  gem "minitest-rails", github: "tsmartt/minitest-rails", branch: "rails71"
```
Comment (thypon) [inline Gemfile:227]: Should this one live in `brave-intl` instead?

File: config/application.rb [hunk 5]
```diff
@@ -11,7 +11,11 @@
 module Publishers
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 7.0
+    config.load_defaults 7.1
+
+    config.pub_secrets = config_for(:secrets) # this line loads the config/secrets.yml file and store it in this namespace
+    # Raise error when a before_action's only/except options reference missing actions
+    config.action_controller.raise_on_missing_callback_actions = false
```
Comment (thypon) [inline config/application.rb:18]: should we fix the missing callbacks instead?

File: lib/publishers/fetch.rb [hunk 6]
```diff
@@ -25,6 +26,7 @@ def fetch(uri:, limit: 10, follow_all_redirects: false, follow_local_redirects:
 
         host = request["host"]
       end
+      response = SsrfFilter.get(uri, max_redirects: limit, request_proc: request_proc, http_options: {open_timeout: 8})
```
Comment (thypon) [inline lib/publishers/fetch.rb:29]: needed?

File: Gemfile (hunk 4 above)
Comment (tsmartt) [reply Gemfile:227]: Sure, I'll as IT to fork

File: Gemfile (hunk 3 above)
Comment (tsmartt) [reply Gemfile:104]: done

File: lib/publishers/fetch.rb (hunk 6 above)
Comment (tsmartt) [reply lib/publishers/fetch.rb:29]: This is how we've always done it for our protection

File: config/application.rb (hunk 5 above)
Comment (tsmartt) [reply config/application.rb:18]: Opened an issue to address in future PR https://github.com/brave-intl/creators-private-issues/issues/1664

File: Gemfile (hunk 4 above)
Comment (tsmartt) [reply Gemfile]: This is done https://github.com/brave-intl/minitest-rails/

**Comment by thypon** [comment]
security gtg, tests fail though!

**Pull request merged**